src/main/java/de/tum/cit/aet/artemis/buildagent/service/SharedQueueProcessingService.java

```diff
@@ -152,14 +171,14 @@ public void init() {
 
         ITopic<String> pauseBuildAgentTopic = hazelcastInstance.getTopic("pauseBuildAgentTopic");
         pauseBuildAgentTopic.addMessageListener(message -> {
-            if (message.getMessageObject().equals(hazelcastInstance.getCluster().getLocalMember().getAddress().toString())) {
+            if (Objects.equals(message.getMessageObject(), buildAgentShortName)) {
```

Comment:
`buildAgentShortName` will never be null
```suggestion
            if (buildAgentShortName.equals(message.getMessageObject())) {
```